File: theme-tests/showcase/preview.js
// HiAstra / JavaScript — expressive code, calm surroundings
const HEX_COLOR = /^#[0-9a-f]{6}$/i;
const settings = { name: "HiAstra", limit: 3, vibrant: true };

/** Local sample data: no network request or API key required. */
async function loadAccents() {
  return [
    { name: "Orbit", color: "#6FA8FF", enabled: true },
    { name: "Aurora", color: "#72E0B2", enabled: true },
    { name: "Flare", color: "#F08ACB", enabled: false },
  ];
}

class ThemePreview {
  constructor(options = settings) {
    this.options = options;
  }

  async render() {
    const accents = await loadAccents();
    const visible = accents.filter(({ color, enabled }) =>
      enabled && HEX_COLOR.test(color)
    );

    for (const [index, accent] of visible.entries()) {
      if (index >= this.options.limit) break;
      console.log(`${this.options.name}: ${accent.name}`);
    }

    return visible.map((accent) => ({ ...accent, selected: false }));
  }
}

new ThemePreview().render().catch(console.error);

File: theme-tests/showcase/preview.py
"""HiAstra / Python — decorators, annotations, and readable logic."""
from dataclasses import dataclass
from re import fullmatch

MAX_SWATCHES = 3


@dataclass(frozen=True)
class Accent:
    name: str
    color: str
    enabled: bool = True

    def label(self, index: int) -> str:
        return f"{index:02d} / {self.name} / {self.color.upper()}"


def preview(accents: list[Accent]) -> list[str]:
    """Select valid colors without changing the original palette."""
    labels = []
    for index, accent in enumerate(accents, start=1):
        if not accent.enabled:
            continue
        if fullmatch(r"#[0-9a-fA-F]{6}", accent.color):
            labels.append(accent.label(index))
    return labels[:MAX_SWATCHES]


palette = [
    Accent("Orbit", "#6FA8FF"),
    Accent("Aurora", "#72E0B2"),
    Accent("Flare", "#F08ACB", enabled=False),
]
print("\n".join(preview(palette)))
